vector/src/main/java/im/vector/app/features/login/LoginViewModel.kt

```diff
@@ -108,32 +120,34 @@ class LoginViewModel @AssistedInject constructor(
 
     override fun handle(action: LoginAction) {
         when (action) {
-            is LoginAction.UpdateServerType           -> handleUpdateServerType(action)
-            is LoginAction.UpdateSignMode             -> handleUpdateSignMode(action)
-            is LoginAction.InitWith                   -> handleInitWith(action)
-            is LoginAction.UpdateHomeServer           -> handleUpdateHomeserver(action).also { lastAction = action }
-            is LoginAction.LoginOrRegister            -> handleLoginOrRegister(action).also { lastAction = action }
-            is LoginAction.LoginWithToken             -> handleLoginWithToken(action)
-            is LoginAction.WebLoginSuccess            -> handleWebLoginSuccess(action)
-            is LoginAction.ResetPassword              -> handleResetPassword(action)
+            is LoginAction.UpdateServerType -> handleUpdateServerType(action)
+            is LoginAction.UpdateSignMode -> handleUpdateSignMode(action)
+            is LoginAction.InitWith -> handleInitWith(action)
+            is LoginAction.UpdateHomeServer -> handleUpdateHomeserver(action).also { lastAction = action }
+            is LoginAction.LoginOrRegister -> handleLoginOrRegister(action).also { lastAction = action }
+            is LoginAction.LoginWithToken -> handleLoginWithToken(action)
+            is LoginAction.WebLoginSuccess -> handleWebLoginSuccess(action)
+            is LoginAction.ResetPassword -> handleResetPassword(action)
             is LoginAction.ResetPasswordMailConfirmed -> handleResetPasswordMailConfirmed()
-            is LoginAction.RegisterAction             -> handleRegisterAction(action)
-            is LoginAction.ResetAction                -> handleResetAction(action)
+            is LoginAction.RegisterAction -> handleRegisterAction(action)
+            is LoginAction.ResetAction -> handleResetAction(action)
             is LoginAction.SetupSsoForSessionRecovery -> handleSetupSsoForSessionRecovery(action)
-            is LoginAction.UserAcceptCertificate      -> handleUserAcceptCertificate(action)
-            is LoginAction.PostViewEvent              -> _viewEvents.post(action.viewEvent)
+            is LoginAction.UserAcceptCertificate -> handleUserAcceptCertificate(action)
+            is LoginAction.PostViewEvent -> _viewEvents.post(action.viewEvent)
         }.exhaustive
     }
 
     private fun handleUserAcceptCertificate(action: LoginAction.UserAcceptCertificate) {
         // It happen when we get the login flow, or during direct authentication.
         // So alter the homeserver config and retrieve again the login flow
         when (val finalLastAction = lastAction) {
-            is LoginAction.UpdateHomeServer ->
+            is LoginAction.UpdateHomeServer -> {
+                rememberHomeServer(finalLastAction.homeServerUrl)
```

Comment:
Should it be better to remember the homeserver only if it appears to be valid? i.e. once the user actually creates a session?

Reply:
Done